File: server-scala-play-server/app/api/PurchaseOrderApi.scala
package api

import play.api.libs.json._
import model.BigDecimal
import model.OrderResponse
import model.SirqulResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
trait PurchaseOrderApi {
  /**
    * Create Order
    * Creates a new purchase with some number of items associated with it. The purchase is added to the order that was created
    * @param appKey The application requesting the purchase
    * @param cart &#x60;&#x60;&#x60;json [   { \&quot;orderItemType\&quot;: \&quot;OFFER\&quot;, \&quot;orderItemId\&quot;: 234, \&quot;orderCustomType\&quot;: \&quot;OfferLocation\&quot;, \&quot;orderCustomId\&quot;: 123, \&quot;retailerLocationId\&quot;: 1234, \&quot;quantity\&quot;: 2 },   { \&quot;orderItemType\&quot;: \&quot;OFFER\&quot;, \&quot;orderItemId\&quot;: 235, \&quot;quantity\&quot;: 2 },   { \&quot;orderItemType\&quot;: \&quot;CUSTOM\&quot;, \&quot;amount\&quot;: 10.50, \&quot;orderCustomType\&quot;: \&quot;ServiceFee\&quot; },   { \&quot;orderItemType\&quot;: \&quot;CUSTOM\&quot;, \&quot;amount\&quot;: 25.10, \&quot;quantity\&quot;: 2, \&quot;orderCustomType\&quot;: \&quot;Hat\&quot;, \&quot;orderCustomId\&quot;: 123 } ] &#x60;&#x60;&#x60; 
    * @param deviceId The device id (deviceId or accountId required)
    * @param accountId The account id of the user (deviceId or accountId required)
    * @param description A description of the purchase
    * @param currencyType Determines the method of purchasing offer.  * VOID: used for when there is no payment involved (ie. for updating the status of a purchase order)  * CASH: use card on file  * POINTS: use account balance  * TICKETS: use tickets  * REFUND: not allowed for a create, requires a paymentTransactionId.  * CREDIT: add to the account balance  * RELOAD: charge a credit card then add to the account balance  
    * @param paymentMethodId Use a specific payment method (CASH), if not provided use default
    * @param externalOrderId Store identifier from external system
    * @param externalPaymentId Store identifier from external system
    * @param remoteRefType Remote Reference type
    * @param externalDate External Date
    * @param promoCode The Promo Code
    */
  def createOrder(version: BigDecimal, appKey: String, cart: String, deviceId: Option[String], accountId: Option[Long], description: Option[String], currencyType: Option[String], paymentMethodId: Option[Long], externalOrderId: Option[String], externalPaymentId: Option[String], remoteRefType: Option[String], externalDate: Option[Long], promoCode: Option[String]): OrderResponse

  /**
    * Delete Order
    * Removes the transaction from the wallet by setting the deleted date to the current date/time.  Requires a valid account and transactionId.
    * @param orderId Order Id
    * @param deviceId The device id (deviceId or accountId required)
    * @param accountId The account id of the user (deviceId or accountId required)
    */
  def deleteOrder(version: BigDecimal, orderId: Long, deviceId: Option[String], accountId: Option[Long]): SirqulResponse

  /**
    * Get Order
    * Get an order record
    * @param deviceId The device id (deviceId or accountId required)
    * @param accountId The account id of the user (deviceId or accountId required)
    * @param orderId The order id to get details of, either orderId or externalOrderId must be provided
    * @param externalOrderId The external order id to get details of, either orderId or externalOrderId must be provided
    */
  def getOrder(version: BigDecimal, deviceId: Option[String], accountId: Option[Long], orderId: Option[Long], externalOrderId: Option[String]): OrderResponse

  /**
    * Preview Order
    * Previews a purchase to see the total cost before making it.
    * @param appKey The application requesting the purchase
    * @param cart A JSON list of items to purchase
    * @param deviceId The device id (deviceId or accountId required)
    * @param accountId The account id of the user (deviceId or accountId required)
    * @param description A description of the purchase
    * @param currencyType Determines the method of purchasing offer. &#x60;&#x60;&#x60;json {   \&quot;VOID\&quot;: \&quot;used for when there is no payment involved (ie. for updating the status of a purchase order)\&quot;,   \&quot;CASH\&quot;: \&quot;use card on file\&quot;,   \&quot;POINTS\&quot;: \&quot;use account balance\&quot;,   \&quot;TICKETS\&quot;: \&quot;use tickets\&quot;,   \&quot;REFUND\&quot;: \&quot;not allowed for a create, requires a paymentTransactionId\&quot;,   \&quot;CREDIT\&quot;: \&quot;add to the account balance\&quot;,   \&quot;RELOAD\&quot;: \&quot;charge a credit card then add to the account balance\&quot; } &#x60;&#x60;&#x60; 
    * @param paymentMethodId Use a specific payment method (CASH), if not provided use default
    * @param externalOrderId Store identifier from external system
    * @param externalPaymentId Store identifier from external system
    * @param remoteRefType Remote Reference type
    * @param externalDate External Date
    * @param promoCode The Promo Code
    */
  def previewOrder(version: BigDecimal, appKey: String, cart: String, deviceId: Option[String], accountId: Option[Long], description: Option[String], currencyType: Option[String], paymentMethodId: Option[Long], externalOrderId: Option[String], externalPaymentId: Option[String], remoteRefType: Option[String], externalDate: Option[Long], promoCode: Option[String]): OrderResponse

  /**
    * Search Orders
    * Search on active orders by customer
    * @param appKey The application requesting the purchase
    * @param deviceId The device id (deviceId or accountId required)
    * @param accountId The account id of the user (deviceId or accountId required)
    * @param start The start index for pagination
    * @param limit The limit for pagination
    * @param descending Determines whether to return the resulting list in descending or ascending order
    * @param activeOnly Only return active orders
    * @param ignoreCustomerFilter Determines whether to ignore the customer filter (requires an Admin/Exec account)
    * @param orderItemTypes Filter results by orderItemTypes
    * @param orderItemIds Filter results by orderItemIds
    * @param orderCustomTypes Filter results by orderCustomTypes
    * @param orderCustomIds Filter results by orderCustomIds
    * @param sortField Determines what to sort the results by
    * @param offerTypes Filter results by offer type
    * @param specialOfferTypes Filter results by special offer type
    * @param categoryIds Filter results by category Ids
    * @param filterIds Filter results by filter Ids
    * @param offerAudienceIds Filter results by offer audience Ids
    * @param transactionAudienceIds Filter results by transaction audience Ids
    * @param offerIds Filter results by offer Ids
    * @param offerLocationIds Filter results by offer location Ids
    * @param retailerIds Filter results by retailer Ids
    * @param retailerLocationIds Filter results by retailer location Ids
    * @param statuses Filter results by offer transaction status (0: non-redeemable, 1: redeemable, 2: redeemed, 3: user deleted, 4: unclaimed/rewarded, 5: membership active, 6: membership cancelled)
    * @param keyword The keyword to search for
    * @param redeemableStartDate Filter results by the offer redeemable date
    * @param redeemableEndDate Filter results by the offer redeemable date
    * @param startedSince Filter results by the offer start date
    * @param startedBefore Filter results by the offer start date
    * @param endedSince Filter results by the offer end date
    * @param endedBefore Filter results by the offer end date
    */
  def searchOrders(version: BigDecimal, appKey: String, deviceId: Option[String], accountId: Option[Long], start: Option[Int], limit: Option[Int], descending: Option[Boolean], activeOnly: Option[Boolean], ignoreCustomerFilter: Option[Boolean], orderItemTypes: Option[String], orderItemIds: Option[String], orderCustomTypes: Option[String], orderCustomIds: Option[String], sortField: Option[String], offerTypes: Option[String], specialOfferTypes: Option[String], categoryIds: Option[String], filterIds: Option[String], offerAudienceIds: Option[String], transactionAudienceIds: Option[String], offerIds: Option[String], offerLocationIds: Option[String], retailerIds: Option[String], retailerLocationIds: Option[String], statuses: Option[String], keyword: Option[String], redeemableStartDate: Option[Long], redeemableEndDate: Option[Long], startedSince: Option[Long], startedBefore: Option[Long], endedSince: Option[Long], endedBefore: Option[Long]): List[OrderResponse]

  /**
    * Update Order
    * Updates new purchase with some number of items associated with it. The orderId provided is used to retrieve the record and the payment is added to it.
    * @param orderId The order to add the purchase to, leave null for new order.
    * @param appKey The application requesting the purchase
    * @param cart &#x60;&#x60;&#x60;json [   { \&quot;orderItemType\&quot;: \&quot;OFFER\&quot;, \&quot;orderItemId\&quot;: 234, \&quot;orderCustomType\&quot;: \&quot;OfferLocation\&quot;, \&quot;orderCustomId\&quot;: 123, \&quot;retailerLocationId\&quot;: 1234, \&quot;quantity\&quot;: 2 },   { \&quot;orderItemType\&quot;: \&quot;OFFER\&quot;, \&quot;orderItemId\&quot;: 235, \&quot;quantity\&quot;: 2 },   { \&quot;orderItemType\&quot;: \&quot;CUSTOM\&quot;, \&quot;amount\&quot;: 10.50, \&quot;orderCustomType\&quot;: \&quot;ServiceFee\&quot; },   { \&quot;orderItemType\&quot;: \&quot;CUSTOM\&quot;, \&quot;amount\&quot;: 25.10, \&quot;quantity\&quot;: 2, \&quot;orderCustomType\&quot;: \&quot;Hat\&quot;, \&quot;orderCustomId\&quot;: 123 } ] &#x60;&#x60;&#x60; 
    * @param deviceId The device id (deviceId or accountId required)
    * @param accountId The account id of the user (deviceId or accountId required)
    * @param paymentTransactionId The payment transaction to apply the refund to
    * @param description A description of the purchase
    * @param currencyType Determines the method of purchasing offer.  * VOID: used for when there is no payment involved (ie. for updating the status of a purchase order)  * CASH: use card on file  * POINTS: use account balance  * TICKETS: use tickets  * REFUND: not allowed for a create, requires a paymentTransactionId.  * CREDIT: add to the account balance  * RELOAD: charge a credit card then add to the account balance  
    * @param paymentMethodId Use a specific payment method (CASH), if not provided use default
    * @param externalPaymentId Store identifier from external system
    * @param externalDate External Date
    */
  def updateOrder(version: BigDecimal, orderId: Long, appKey: String, cart: String, deviceId: Option[String], accountId: Option[Long], paymentTransactionId: Option[Long], description: Option[String], currencyType: Option[String], paymentMethodId: Option[Long], externalPaymentId: Option[String], externalDate: Option[Long]): OrderResponse
}

File: server-ruby-sinatra/api/purchase_order_api.rb
require 'json'


MyApp.add_route('POST', '/api/{version}/order/create', {
  "resourcePath" => "/PurchaseOrder",
  "summary" => "Create Order",
  "nickname" => "create_order",
  "responseClass" => "OrderResponse",
  "endpoint" => "/api/{version}/order/create",
  "notes" => "Creates a new purchase with some number of items associated with it. The purchase is added to the order that was created",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The device id (deviceId or accountId required)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "account_id",
      "description" => "The account id of the user (deviceId or accountId required)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "app_key",
      "description" => "The application requesting the purchase",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "description",
      "description" => "A description of the purchase",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "currency_type",
      "description" => "Determines the method of purchasing offer.  * VOID: used for when there is no payment involved (ie. for updating the status of a purchase order)  * CASH: use card on file  * POINTS: use account balance  * TICKETS: use tickets  * REFUND: not allowed for a create, requires a paymentTransactionId.  * CREDIT: add to the account balance  * RELOAD: charge a credit card then add to the account balance  ",
      "dataType" => "String",
      "allowableValues" => "[VOID, CASH, POINTS, TICKETS, REFUND, CREDIT, RELOAD]",
      "defaultValue" => "'CASH'",
      "paramType" => "query",
    },
    {
      "name" => "cart",
      "description" => "&#x60;&#x60;&#x60;json [   { \&quot;orderItemType\&quot;: \&quot;OFFER\&quot;, \&quot;orderItemId\&quot;: 234, \&quot;orderCustomType\&quot;: \&quot;OfferLocation\&quot;, \&quot;orderCustomId\&quot;: 123, \&quot;retailerLocationId\&quot;: 1234, \&quot;quantity\&quot;: 2 },   { \&quot;orderItemType\&quot;: \&quot;OFFER\&quot;, \&quot;orderItemId\&quot;: 235, \&quot;quantity\&quot;: 2 },   { \&quot;orderItemType\&quot;: \&quot;CUSTOM\&quot;, \&quot;amount\&quot;: 10.50, \&quot;orderCustomType\&quot;: \&quot;ServiceFee\&quot; },   { \&quot;orderItemType\&quot;: \&quot;CUSTOM\&quot;, \&quot;amount\&quot;: 25.10, \&quot;quantity\&quot;: 2, \&quot;orderCustomType\&quot;: \&quot;Hat\&quot;, \&quot;orderCustomId\&quot;: 123 } ] &#x60;&#x60;&#x60; ",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "payment_method_id",
      "description" => "Use a specific payment method (CASH), if not provided use default",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_order_id",
      "description" => "Store identifier from external system",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_payment_id",
      "description" => "Store identifier from external system",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "remote_ref_type",
      "description" => "Remote Reference type",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_date",
      "description" => "External Date",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "promo_code",
      "description" => "The Promo Code",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/api/{version}/order/delete', {
  "resourcePath" => "/PurchaseOrder",
  "summary" => "Delete Order",
  "nickname" => "delete_order",
  "responseClass" => "SirqulResponse",
  "endpoint" => "/api/{version}/order/delete",
  "notes" => "Removes the transaction from the wallet by setting the deleted date to the current date/time.  Requires a valid account and transactionId.",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The device id (deviceId or accountId required)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "account_id",
      "description" => "The account id of the user (deviceId or accountId required)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "order_id",
      "description" => "Order Id",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/{version}/order/get', {
  "resourcePath" => "/PurchaseOrder",
  "summary" => "Get Order",
  "nickname" => "get_order",
  "responseClass" => "OrderResponse",
  "endpoint" => "/api/{version}/order/get",
  "notes" => "Get an order record",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The device id (deviceId or accountId required)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "account_id",
      "description" => "The account id of the user (deviceId or accountId required)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "order_id",
      "description" => "The order id to get details of, either orderId or externalOrderId must be provided",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_order_id",
      "description" => "The external order id to get details of, either orderId or externalOrderId must be provided",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/api/{version}/order/preview', {
  "resourcePath" => "/PurchaseOrder",
  "summary" => "Preview Order",
  "nickname" => "preview_order",
  "responseClass" => "OrderResponse",
  "endpoint" => "/api/{version}/order/preview",
  "notes" => "Previews a purchase to see the total cost before making it.",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The device id (deviceId or accountId required)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "account_id",
      "description" => "The account id of the user (deviceId or accountId required)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "app_key",
      "description" => "The application requesting the purchase",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "description",
      "description" => "A description of the purchase",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "currency_type",
      "description" => "Determines the method of purchasing offer. &#x60;&#x60;&#x60;json {   \&quot;VOID\&quot;: \&quot;used for when there is no payment involved (ie. for updating the status of a purchase order)\&quot;,   \&quot;CASH\&quot;: \&quot;use card on file\&quot;,   \&quot;POINTS\&quot;: \&quot;use account balance\&quot;,   \&quot;TICKETS\&quot;: \&quot;use tickets\&quot;,   \&quot;REFUND\&quot;: \&quot;not allowed for a create, requires a paymentTransactionId\&quot;,   \&quot;CREDIT\&quot;: \&quot;add to the account balance\&quot;,   \&quot;RELOAD\&quot;: \&quot;charge a credit card then add to the account balance\&quot; } &#x60;&#x60;&#x60; ",
      "dataType" => "String",
      "allowableValues" => "[VOID, CASH, POINTS, TICKETS, REFUND, CREDIT, RELOAD]",
      "defaultValue" => "'CASH'",
      "paramType" => "query",
    },
    {
      "name" => "cart",
      "description" => "A JSON list of items to purchase",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "payment_method_id",
      "description" => "Use a specific payment method (CASH), if not provided use default",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_order_id",
      "description" => "Store identifier from external system",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_payment_id",
      "description" => "Store identifier from external system",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "remote_ref_type",
      "description" => "Remote Reference type",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_date",
      "description" => "External Date",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "promo_code",
      "description" => "The Promo Code",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/{version}/order/search', {
  "resourcePath" => "/PurchaseOrder",
  "summary" => "Search Orders",
  "nickname" => "search_orders",
  "responseClass" => "Array<OrderResponse>",
  "endpoint" => "/api/{version}/order/search",
  "notes" => "Search on active orders by customer",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The device id (deviceId or accountId required)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "account_id",
      "description" => "The account id of the user (deviceId or accountId required)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "app_key",
      "description" => "The application requesting the purchase",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "start",
      "description" => "The start index for pagination",
      "dataType" => "Integer",
      "allowableValues" => "",
      "defaultValue" => "0",
      "paramType" => "query",
    },
    {
      "name" => "limit",
      "description" => "The limit for pagination",
      "dataType" => "Integer",
      "allowableValues" => "",
      "defaultValue" => "20",
      "paramType" => "query",
    },
    {
      "name" => "descending",
      "description" => "Determines whether to return the resulting list in descending or ascending order",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "defaultValue" => "true",
      "paramType" => "query",
    },
    {
      "name" => "active_only",
      "description" => "Only return active orders",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "defaultValue" => "false",
      "paramType" => "query",
    },
    {
      "name" => "ignore_customer_filter",
      "description" => "Determines whether to ignore the customer filter (requires an Admin/Exec account)",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "defaultValue" => "false",
      "paramType" => "query",
    },
    {
      "name" => "order_item_types",
      "description" => "Filter results by orderItemTypes",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "order_item_ids",
      "description" => "Filter results by orderItemIds",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "order_custom_types",
      "description" => "Filter results by orderCustomTypes",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "order_custom_ids",
      "description" => "Filter results by orderCustomIds",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "sort_field",
      "description" => "Determines what to sort the results by",
      "dataType" => "String",
      "allowableValues" => "",
      "defaultValue" => "'ID'",
      "paramType" => "query",
    },
    {
      "name" => "offer_types",
      "description" => "Filter results by offer type",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "special_offer_types",
      "description" => "Filter results by special offer type",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "category_ids",
      "description" => "Filter results by category Ids",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "filter_ids",
      "description" => "Filter results by filter Ids",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "offer_audience_ids",
      "description" => "Filter results by offer audience Ids",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "transaction_audience_ids",
      "description" => "Filter results by transaction audience Ids",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "offer_ids",
      "description" => "Filter results by offer Ids",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "offer_location_ids",
      "description" => "Filter results by offer location Ids",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "retailer_ids",
      "description" => "Filter results by retailer Ids",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "retailer_location_ids",
      "description" => "Filter results by retailer location Ids",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "statuses",
      "description" => "Filter results by offer transaction status (0: non-redeemable, 1: redeemable, 2: redeemed, 3: user deleted, 4: unclaimed/rewarded, 5: membership active, 6: membership cancelled)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "keyword",
      "description" => "The keyword to search for",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "redeemable_start_date",
      "description" => "Filter results by the offer redeemable date",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "redeemable_end_date",
      "description" => "Filter results by the offer redeemable date",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "started_since",
      "description" => "Filter results by the offer start date",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "started_before",
      "description" => "Filter results by the offer start date",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "ended_since",
      "description" => "Filter results by the offer end date",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "ended_before",
      "description" => "Filter results by the offer end date",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/api/{version}/order/update', {
  "resourcePath" => "/PurchaseOrder",
  "summary" => "Update Order",
  "nickname" => "update_order",
  "responseClass" => "OrderResponse",
  "endpoint" => "/api/{version}/order/update",
  "notes" => "Updates new purchase with some number of items associated with it. The orderId provided is used to retrieve the record and the payment is added to it.",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The device id (deviceId or accountId required)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "account_id",
      "description" => "The account id of the user (deviceId or accountId required)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "order_id",
      "description" => "The order to add the purchase to, leave null for new order.",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "payment_transaction_id",
      "description" => "The payment transaction to apply the refund to",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "app_key",
      "description" => "The application requesting the purchase",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "description",
      "description" => "A description of the purchase",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "currency_type",
      "description" => "Determines the method of purchasing offer.  * VOID: used for when there is no payment involved (ie. for updating the status of a purchase order)  * CASH: use card on file  * POINTS: use account balance  * TICKETS: use tickets  * REFUND: not allowed for a create, requires a paymentTransactionId.  * CREDIT: add to the account balance  * RELOAD: charge a credit card then add to the account balance  ",
      "dataType" => "String",
      "allowableValues" => "[VOID, CASH, POINTS, TICKETS, REFUND, CREDIT, RELOAD]",
      "defaultValue" => "'CASH'",
      "paramType" => "query",
    },
    {
      "name" => "cart",
      "description" => "&#x60;&#x60;&#x60;json [   { \&quot;orderItemType\&quot;: \&quot;OFFER\&quot;, \&quot;orderItemId\&quot;: 234, \&quot;orderCustomType\&quot;: \&quot;OfferLocation\&quot;, \&quot;orderCustomId\&quot;: 123, \&quot;retailerLocationId\&quot;: 1234, \&quot;quantity\&quot;: 2 },   { \&quot;orderItemType\&quot;: \&quot;OFFER\&quot;, \&quot;orderItemId\&quot;: 235, \&quot;quantity\&quot;: 2 },   { \&quot;orderItemType\&quot;: \&quot;CUSTOM\&quot;, \&quot;amount\&quot;: 10.50, \&quot;orderCustomType\&quot;: \&quot;ServiceFee\&quot; },   { \&quot;orderItemType\&quot;: \&quot;CUSTOM\&quot;, \&quot;amount\&quot;: 25.10, \&quot;quantity\&quot;: 2, \&quot;orderCustomType\&quot;: \&quot;Hat\&quot;, \&quot;orderCustomId\&quot;: 123 } ] &#x60;&#x60;&#x60; ",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "payment_method_id",
      "description" => "Use a specific payment method (CASH), if not provided use default",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_payment_id",
      "description" => "Store identifier from external system",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_date",
      "description" => "External Date",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

